src_c/polygon.c

```diff
@@ -468,6 +468,31 @@ pg_polygon_move(pgPolygonObject *self, PyObject *const *args, Py_ssize_t nargs)
     return tmp;
 }
 
+static PyObject *
+pg_polygon_as_segments(pgPolygonObject *self) {
+    PyObject *list = PyList_New(0);
```

Comment:
Given that you know how many vertices you have in a polygon you can preallocate a list of size `verts_num` with `PyObject *list = PyList_New(self->polygon.verts_num);`. Later on for appending to a preallocated list you must use the `PyList_SET_ITEM` macro.